tests/consensus_tests/test_resharding.py

```diff
@@ -2,991 +2,307 @@
+    try_requests(peer_uris[0], 400, [
+        start_resharding,
+        lambda peer_uri: start_resharding(peer_uri, direction="down"),
+        commit_write_hashring,
+        finish_resharding,
+    ])
 
-    # Match all points on all nodes exactly
-    data = []
-    for uri in peer_api_uris:
-        r = requests.post(
-            f"{uri}/collections/{COLLECTION_NAME}/points/scroll", json={
-                "limit": 999999999,
-                "with_vectors": True,
-                "with_payload": True,
-            }
-        )
-        assert_http_ok(r)
-        data.append(r.json()["result"])
-    check_data_consistency(data)
-
-
-# Test resharding shard balancing.
-#
-# Sets up a 3 node cluster and a collection with 1 shard and 2 replicas.
-# Performs resharding 7 times and asserts the shards replicas are evenly
-# balanced across all nodes.
-#
-# In this case the replicas are balanced on the second and third node. The first
-# node has all shards because we explicitly set it as shard target all the time.
-@pytest.mark.skip(reason="moving resharding driver to external service")
-def test_resharding_balance(tmp_path: pathlib.Path):
-    assert_project_root()
+    # Commit read hashring
+    resp = commit_read_hashring(peer_uris[0])
+    assert_http_ok(resp)
 
-    num_points = 100
+    # Check that we can't (re)apply unexpected resharding state transitions or abort resharding
+    try_requests(peer_uris[0], 400, [
+        start_resharding,
+        lambda peer_uri: start_resharding(peer_uri, direction="down"),
+        commit_read_hashring,
+        finish_resharding,
+        abort_resharding,
+    ])
+
+    # Commit write hashring
+    resp = commit_write_hashring(peer_uris[0])
+    assert_http_ok(resp)
 
-    # Prevent optimizers messing with point counts
-    env={
-        "QDRANT__STORAGE__OPTIMIZERS__INDEXING_THRESHOLD_KB": "0",
-    }
+    # Check that we can't (re)apply unexpected resharding state transitions or abort resharding
+    try_requests(peer_uris[0], 400, [
+        start_resharding,
+        lambda peer_uri: start_resharding(peer_uri, direction="down"),
+        commit_read_hashring,
+        commit_write_hashring,
+        abort_resharding,
+    ])
+
+    # Finish resharding
+    resp = finish_resharding(peer_uris[0])
+    assert_http_ok(resp)
 
-    peer_api_uris, _peer_dirs, _bootstrap_uri = start_cluster(tmp_path, 3, None, extra_env=env)
-    first_peer_id = get_cluster_info(peer_api_uris[0])['peer_id']
+    # Wait for resharding to finish
+    wait_for_collection_resharding_operations_count(peer_uris[0], COLLECTION_NAME, 0)
 
-    # Create collection, insert points
-    create_collection(peer_api_uris[0], shard_number=1, replication_factor=2)
-    wait_collection_exists_and_active_on_all_peers(
-        collection_name=COLLECTION_NAME,
-        peer_api_uris=peer_api_uris,
-    )
-    upsert_random_points(peer_api_uris[0], num_points)
+def test_resharding_abort(tmp_path: pathlib.Path):
```

Comment:
Add another test to assert we cannot abort after committing hash rings?

Reply:
Checked in previous test: `try_requests` calls after `commit_read_hashring` and `commit_write_hashring` include `abort_resharding`.